Libraries/PeriphDrivers/Source/SPI/spi_ai87.c

```diff
@@ -403,34 +403,19 @@ int MXC_SPI_MasterTransactionDMA(mxc_spi_req_t *req)
     spi_num = MXC_SPI_GET_IDX(req->spi);
     MXC_ASSERT(spi_num >= 0);
 
-    if (req->txData != NULL) {
-        switch (spi_num) {
-        case 0:
-            reqselTx = MXC_DMA_REQUEST_SPI1TX;
-            break;
-
-        case 1:
-            reqselTx = MXC_DMA_REQUEST_SPI0TX;
-            break;
-
-        default:
-            return E_BAD_PARAM;
-        }
-    }
-
-    if (req->rxData != NULL) {
-        switch (spi_num) {
-        case 0:
-            reqselRx = MXC_DMA_REQUEST_SPI1RX;
-            break;
-
-        case 1:
-            reqselRx = MXC_DMA_REQUEST_SPI0RX;
-            break;
-
-        default:
-            return E_BAD_PARAM;
-        }
+    switch (spi_num) {
+    case 0:
+        reqselTx = MXC_DMA_REQUEST_SPI1TX;
```

Comment:
I believe the original implementation was correct. SPI0 and SPI1 are swapped in several places as an unfortunate result of incorrect documentation, and it's been stuck like that ever since to prevent breaking projects... It's why we have messes like this for the ME17, AI85, and AI87. 
![image](https://github.com/analogdevicesinc/msdk/assets/75494566/3c080c73-420e-443d-b8d5-4b675004f450)

I say we fix this even if it might break existing projects. Usually, instance 0 is mapped to the lowest peripheral address, and then everything goes in ascending order:
1. SPI0 (APB) - 0x40046000
2. SPI1 (APB) - 0x40047000
3. SPI[n] (APB) - 0x4004m000 
4. SPI[n+1] (AHB) - 0x400BE000
5. SPI[n+2] (AHB) - 0x400BE400

However, the ME17, AI85, and AI87 doesn't follow this:
1. SPI0 (AHB) - 0x400BE000
2. SPI1 (APB) - 0x40046000

This has caused confusion, but it'll be a major breaking change to swap the SPI instance names. Instead, we can fix the `MXC_SPI_GET_*` definitions in the {device}.h file where SPI0 would be associated with index 0, and SPI1 with index 1. This would still be a breaking change, but it would at least help keep all the indexing aligned with the correct SPI names.

What do you think? @Jake-Carter & @lorne-maxim